tests/admin/test-class-admin-editor-specific-replace-vars.php

```diff
@@ -8,247 +8,250 @@
+
+		$this->assertEquals( 'page', $this->class_instance->determine_for_post( $post ) );
+
+		// Revert the options their original values.
+		update_option( 'show_on_front', $show_on_front );
+		update_option( 'page_on_front', $page_on_front );
+	}
+
+	/**
+	 * Tests that determine_for_post can detect a page.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_page() {
+		$post = $this->create_and_get_with_post_type( 'page' );
+
+		$this->assertEquals( 'page', $this->class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Tests that determine_for_post can detect a post.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_post() {
+		$post = $this->create_and_get_with_post_type( 'post' );
+
+		$this->assertEquals( 'post', $this->class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Tests that determine_for_post can detect a custom post type.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_custom_post_type() {
+		$post = $this->create_and_get_with_post_type( 'some_plugin_post' );
+
+		$this->assertEquals( 'custom_post_type', $this->class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Tests that determine_for_post_type works for a post.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post_type
+	 */
+	public function test_determine_for_post_type_with_a_post() {
+		$this->assertEquals( 'post', $this->class_instance->determine_for_post_type( 'post' ) );
+	}
+
+	/**
+	 * Tests that the determine_for_post_type fallback works.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post_type
+	 */
+	public function test_determine_for_post_type_with_a_fallback() {
+		$this->assertEquals( 'custom_post_type', $this->class_instance->determine_for_post_type( 'non-existing-post_type' ) );
+	}
+
+	/**
+	 * Tests that the determine_for_post_type custom fallback works.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post_type
+	 */
+	public function test_determine_for_post_type_with_a_custom_fallback() {
+		$this->assertEquals( 'fallback_post_type', $this->class_instance->determine_for_post_type( 'non-existing-post_type', 'fallback_post_type' ) );
+	}
+
+	/**
+	 * Tests that determine_for_archive works for the date_archive.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_archive
+	 */
+	public function test_determine_for_archive_with_author() {
+		$this->assertEquals( 'custom-post-type_archive', $this->class_instance->determine_for_archive( 'author' ) );
+	}
+
+	/**
+	 * Tests that determine_for_archive works for the date_archive.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_archive
+	 */
+	public function test_determine_for_archive_with_date() {
```

Comment:
You might consider using a dataprovider to have on test implementation for `determine_for_archive`